frontend/catalyst/qfunc.py

```diff
@@ -272,7 +271,6 @@ def processing_fn(results):
     total_shots = _get_total_shots(qnode)
 
     new_dev = copy(dev)
-    new_dev._shots = qml.measurements.Shots(1)
     single_shot_qnode.device = new_dev
     single_shot_qnode._set_shots(qml.measurements.Shots(1))  # pylint: disable=protected-access
```

Comment:
Does this method need replacing somehow? Looking at the impl, I guess QNodes were already storing a shots attribute internally even before this change.

Reply:
Yes but before this PR, the interwinding shots information flows among QJITDevice and verification and tracer and qfunc made things quite convoluting... I introduced these modified devices to keep dynamic one shot qfunc consistent end-to-end after the introduction of `set_shots`

Reply:
@dime10 Which method? `qnode._set_shots`?

Reply:
Nice, that's a good simplification actually :)